Support CSS color values that use calc() with non-absolute lengths

weinig opened the pull request:
#### d7aed1e4ddffee252473469567148feb4fa1fac4
<pre>
Support CSS color values that use calc() with non-absolute lengths
<a href="https://bugs.webkit.org/show_bug.cgi?id=278547">https://bugs.webkit.org/show_bug.cgi?id=278547</a>

Reviewed by Darin Adler.

Adds support for CSS colors that use calc() with non-absolute lengths
such as:

    hsl(calc(50deg + (sign(1em - 10px) * 10deg)), 100%, 37.5%, 50%)

These have the special property that their calc() cannot be evaluated
at parse time, as they require conversion data. As such, they need to
preserve and serialize the calc().

To support this (and prepare for upcoming changes to nested colors)
CSSUnresolvedAbsoluteColor has been repurposed as a generic type that
mirrors CSSUnresolvedRelativeColor, and the old functionality moved
to a new CSSUnresolvedAbsoluteResolvedColor. The parser now creates
CSSUnresolvedAbsoluteColor if the calc() cannot be eagerly evaluated,
otherwise, using CSSUnresolvedAbsoluteResolvedColor as it used to.

Additionally, CSSToLengthConversionData is now needed when resolving
CSSUnresolvedColors to StyleColor. Instead of adding yet another
parameter to `createStyleColor`, the parameters are now packaged in
a new CSSUnresolvedStyleColorResolutionState struct (and to match,
CSSUnresolvedColorResolutionContext has been renamed to use the State
suffix).

With CSSToLengthConversionData now available, it is also used for
resolving the percentages in `color-mix()` colors.

* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-color-function-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-color-function.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-color-mix-function-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-color-mix-function.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-hsl-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-hsl.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-hwb-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-hwb.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-lab-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-lab.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-rgb-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-rgb.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-color-function-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-color-function.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-color-mix-function-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-color-mix-function.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-hsl-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-hsl.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-hwb-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-hwb.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-lab-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-lab.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-rgb-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-rgb.html:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/calc/CSSCalcTree+Copy.cpp: Copied from Source/WebCore/platform/calc/CalculationTree+Copy.cpp.
* Source/WebCore/css/calc/CSSCalcTree+Copy.h: Copied from Source/WebCore/css/calc/CSSCalcTree+Serialization.h.
* Source/WebCore/css/calc/CSSCalcTree+Parser.cpp:
(WebCore::CSSCalc::parseAndSimplify):
* Source/WebCore/css/calc/CSSCalcTree+Serialization.cpp:
(WebCore::CSSCalc::serializationForCSS):
(WebCore::CSSCalc::serializeForCSS): Deleted.
* Source/WebCore/css/calc/CSSCalcTree+Serialization.h:
* Source/WebCore/css/calc/CSSCalcTree.cpp:
(WebCore::CSSCalc::operator&lt;&lt;):
* Source/WebCore/css/calc/CSSCalcValue.cpp:
(WebCore::CSSCalcValue::copySimplified const):
(WebCore::CSSCalcValue::requiresConversionData const):
(WebCore::CSSCalcValue::customCSSText const):
(WebCore::CSSCalcValue::dump const):
* Source/WebCore/css/calc/CSSCalcValue.h:
* Source/WebCore/css/color/CSSAbsoluteColorResolver.h:
(WebCore::requiresConversionData):
(WebCore::resolve):
(WebCore::resolveNoConversionDataRequired):
* Source/WebCore/css/color/CSSAbsoluteColorSerialization.h: Copied from Source/WebCore/css/color/CSSUnresolvedColorLayers.h.
(WebCore::serializationForCSSAbsoluteColor):
* Source/WebCore/css/color/CSSColorConversion+ToTypedColor.h:
* Source/WebCore/css/color/CSSColorMixSerialization.cpp:
(WebCore::is50Percent):
(WebCore::sumTo100Percent):
(WebCore::serializeColorMixColor):
(WebCore::serializeColorMix100MinusPercentage):
(WebCore::percentageDoubleValue): Deleted.
* Source/WebCore/css/color/CSSColorMixSerialization.h:
(WebCore::serializationForCSSColorMix):
* Source/WebCore/css/color/CSSRelativeColorResolver.h:
(WebCore::requiresConversionData):
(WebCore::resolve):
(WebCore::resolveNoConversionDataRequired):
* Source/WebCore/css/color/CSSUnresolvedAbsoluteColor.h:
(WebCore::serializationForCSS):
(WebCore::createStyleColor):
(WebCore::createColor):
(WebCore::containsColorSchemeDependentColor):
(WebCore::containsCurrentColor):
* Source/WebCore/css/color/CSSUnresolvedAbsoluteResolvedColor.cpp: Renamed from Source/WebCore/css/color/CSSUnresolvedAbsoluteColor.cpp.
* Source/WebCore/css/color/CSSUnresolvedAbsoluteResolvedColor.h: Copied from Source/WebCore/css/color/CSSUnresolvedAbsoluteColor.h.
* Source/WebCore/css/color/CSSUnresolvedColor.cpp:
* Source/WebCore/css/color/CSSUnresolvedColor.h:
* Source/WebCore/css/color/CSSUnresolvedColorHex.h:
* Source/WebCore/css/color/CSSUnresolvedColorKeyword.cpp:
* Source/WebCore/css/color/CSSUnresolvedColorKeyword.h:
* Source/WebCore/css/color/CSSUnresolvedColorLayers.cpp:
* Source/WebCore/css/color/CSSUnresolvedColorLayers.h:
* Source/WebCore/css/color/CSSUnresolvedColorMix.cpp:
* Source/WebCore/css/color/CSSUnresolvedColorMix.h:
* Source/WebCore/css/color/CSSUnresolvedColorResolutionState.cpp: Renamed from Source/WebCore/css/color/CSSUnresolvedColorResolutionContext.cpp.
* Source/WebCore/css/color/CSSUnresolvedColorResolutionState.h: Renamed from Source/WebCore/css/color/CSSUnresolvedColorResolutionContext.h.
* Source/WebCore/css/color/CSSUnresolvedLightDark.cpp:
* Source/WebCore/css/color/CSSUnresolvedLightDark.h:
* Source/WebCore/css/color/CSSUnresolvedRelativeColor.h:
* Source/WebCore/css/color/CSSUnresolvedStyleColorResolutionState.h: Copied from Source/WebCore/style/ColorFromPrimitiveValue.h.
* Source/WebCore/css/color/StyleColor.cpp:
* Source/WebCore/css/color/StyleColor.h:
* Source/WebCore/css/color/StyleColorLayers.cpp:
* Source/WebCore/css/color/StyleColorLayers.h:
* Source/WebCore/css/color/StyleColorMix.cpp:
* Source/WebCore/css/color/StyleColorMix.h:
* Source/WebCore/css/color/StyleRelativeColor.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Color.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Color.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+RawResolver.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+UnevaluatedCalc.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+UnevaluatedCalc.h:
* Source/WebCore/html/canvas/CanvasStyle.cpp:
* Source/WebCore/platform/calc/CalculationTree+Copy.cpp:
* Source/WebCore/style/ColorFromPrimitiveValue.cpp:
* Source/WebCore/style/ColorFromPrimitiveValue.h:
* Source/WebCore/style/FilterOperationsBuilder.cpp:
* Source/WebCore/style/StyleBuilderState.cpp:

Canonical link: <a href="https://commits.webkit.org/282670@main">https://commits.webkit.org/282670@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f24965421682045aab1fe9d45c077b6f56c2d6a4

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/63886 "11 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/43243 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/16483 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/67908 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/14494 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/66006 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/50941 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/14774 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/51478 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/10021 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/66955 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/40050 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/55306 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/32094 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/36728 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/12691 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/13367 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/58692 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/13018 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/69604 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/7833 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/12555 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/58793 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/7866 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/55403 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/58939 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14132 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/6512 "Passed tests") | | | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/39063 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/40142 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/41325 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/39885 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/css/calc/CSSCalcTree+Copy.cpp:
```diff
@@ -0,0 +1,116 @@
+template<Leaf Op>
+Child copy(const Op&);
+template<typename Op>
+static auto copy(const IndirectNode<Op>&) -> Child;
```
I fine these kinds of declarations so much easier to read on a single line.

darinadler commented on Source/WebCore/css/calc/CSSCalcTree+Serialization.h:
```diff
@@ -31,11 +31,11 @@ namespace WebCore {
 // https://drafts.csswg.org/css-values-4/#serialize-a-math-function
-void serializeForCSS(StringBuilder&, const Tree&);
-String serializeForCSS(const Tree&);
+void serializationForCSS(StringBuilder&, const Tree&);
```
Why isn’t this called something more like `appendCSSSerialization`? Is it important that the `StringBuilder` version be an overload?

weinig replied:
I am not sure there is a good reason other than consistency with some subset of types (I guess mostly color related, so maybe this was just a pattern I made and didn't noticed there was an established precedent elsewhere.

Overloading wasn't the goal, and in fact is kind of a downside when used generically.

darinadler commented on Source/WebCore/css/calc/CSSCalcValue.cpp:
```diff
@@ -95,6 +95,20 @@ Ref<CSSCalcValue> CSSCalcValue::create(CSSCalc::Tree&& tree)
+        .allowNonMatchingUnits = false
+    };
+
+    return CSSCalcValue::create(copyAndSimplify(m_tree, simplificationOptions));
```
Can just call `create` here instead of `CSSCalcValue::create`.

darinadler commented on Source/WebCore/css/color/CSSAbsoluteColorSerialization.h:
```diff
@@ -0,0 +1,68 @@
+namespace WebCore {
+
+class StyleColor;
+class CSSUnresolvedColor;
```
Sort alphabetically, please

weinig replied:
Heh, I looked at that a few times and said, "hey, that is sorted",, but it was just my brain eliminating the "CSS" prefix and tricking me. :)

darinadler commented on Source/WebCore/css/color/CSSColorMixSerialization.cpp:
```diff
@@ -35,9 +38,47 @@
+{
+    auto visitor = WTF::makeVisitor(
+        [](const PercentRaw& a, const PercentRaw& b) {
+            return (a.value + b.value) == 100.0;
```
I don’t think we need parentheses here.

darinadler commented on Source/WebCore/css/color/CSSColorMixSerialization.cpp:
```diff
@@ -35,9 +38,47 @@
+
+bool sumTo100Percent(const StyleColorMix::Component::Percentage& a, const StyleColorMix::Component::Percentage& b)
+{
+    return (a.value + b.value) == 100.0;
```
I don’t think we need parentheses here.

darinadler reviewed (approved):
I looked mostly at the code structure, not correctness. Assuming the test coverage is how we know it’s right.

weinig committed:
Support CSS color values that use calc() with non-absolute lengths
https://bugs.webkit.org/show_bug.cgi?id=278547

Reviewed by Darin Adler.

Adds support for CSS colors that use calc() with non-absolute lengths
such as:

    hsl(calc(50deg + (sign(1em - 10px) * 10deg)), 100%, 37.5%, 50%)

These have the special property that their calc() cannot be evaluated
at parse time, as they require conversion data. As such, they need to
preserve and serialize the calc().

To support this (and prepare for upcoming changes to nested colors)
CSSUnresolvedAbsoluteColor has been repurposed as a generic type that
mirrors CSSUnresolvedRelativeColor, and the old functionality moved
to a new CSSUnresolvedAbsoluteResolvedColor. The parser now creates
CSSUnresolvedAbsoluteColor if the calc() cannot be eagerly evaluated,
otherwise, using CSSUnresolvedAbsoluteResolvedColor as it used to.

Additionally, CSSToLengthConversionData is now needed when resolving
CSSUnresolvedColors to StyleColor. Instead of adding yet another
parameter to `createStyleColor`, the parameters are now packaged in
a new CSSUnresolvedStyleColorResolutionState struct (and to match,
CSSUnresolvedColorResolutionContext has been renamed to use the State
suffix).

With CSSToLengthConversionData now available, it is also used for
resolving the percentages in `color-mix()` colors.

* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-color-function-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-color-function.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-color-mix-function-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-color-mix-function.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-hsl-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-hsl.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-hwb-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-hwb.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-lab-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-lab.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-rgb-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-rgb.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-color-function-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-color-function.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-color-mix-function-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-color-mix-function.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-hsl-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-hsl.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-hwb-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-hwb.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-lab-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-lab.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-rgb-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-rgb.html:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/calc/CSSCalcTree+Copy.cpp: Copied from Source/WebCore/platform/calc/CalculationTree+Copy.cpp.
* Source/WebCore/css/calc/CSSCalcTree+Copy.h: Copied from Source/WebCore/css/calc/CSSCalcTree+Serialization.h.
* Source/WebCore/css/calc/CSSCalcTree+Parser.cpp:
(WebCore::CSSCalc::parseAndSimplify):
* Source/WebCore/css/calc/CSSCalcTree+Serialization.cpp:
(WebCore::CSSCalc::serializationForCSS):
(WebCore::CSSCalc::serializeForCSS): Deleted.
* Source/WebCore/css/calc/CSSCalcTree+Serialization.h:
* Source/WebCore/css/calc/CSSCalcTree.cpp:
(WebCore::CSSCalc::operator<<):
* Source/WebCore/css/calc/CSSCalcValue.cpp:
(WebCore::CSSCalcValue::copySimplified const):
(WebCore::CSSCalcValue::requiresConversionData const):
(WebCore::CSSCalcValue::customCSSText const):
(WebCore::CSSCalcValue::dump const):
* Source/WebCore/css/calc/CSSCalcValue.h:
* Source/WebCore/css/color/CSSAbsoluteColorResolver.h:
(WebCore::requiresConversionData):
(WebCore::resolve):
(WebCore::resolveNoConversionDataRequired):
* Source/WebCore/css/color/CSSAbsoluteColorSerialization.h: Copied from Source/WebCore/css/color/CSSUnresolvedColorLayers.h.
(WebCore::serializationForCSSAbsoluteColor):
* Source/WebCore/css/color/CSSColorConversion+ToTypedColor.h:
* Source/WebCore/css/color/CSSColorMixSerialization.cpp:
(WebCore::is50Percent):
(WebCore::sumTo100Percent):
(WebCore::serializeColorMixColor):
(WebCore::serializeColorMix100MinusPercentage):
(WebCore::percentageDoubleValue): Deleted.
* Source/WebCore/css/color/CSSColorMixSerialization.h:
(WebCore::serializationForCSSColorMix):
* Source/WebCore/css/color/CSSRelativeColorResolver.h:
(WebCore::requiresConversionData):
(WebCore::resolve):
(WebCore::resolveNoConversionDataRequired):
* Source/WebCore/css/color/CSSUnresolvedAbsoluteColor.h:
(WebCore::serializationForCSS):
(WebCore::createStyleColor):
(WebCore::createColor):
(WebCore::containsColorSchemeDependentColor):
(WebCore::containsCurrentColor):
* Source/WebCore/css/color/CSSUnresolvedAbsoluteResolvedColor.cpp: Renamed from Source/WebCore/css/color/CSSUnresolvedAbsoluteColor.cpp.
* Source/WebCore/css/color/CSSUnresolvedAbsoluteResolvedColor.h: Copied from Source/WebCore/css/color/CSSUnresolvedAbsoluteColor.h.
* Source/WebCore/css/color/CSSUnresolvedColor.cpp:
* Source/WebCore/css/color/CSSUnresolvedColor.h:
* Source/WebCore/css/color/CSSUnresolvedColorHex.h:
* Source/WebCore/css/color/CSSUnresolvedColorKeyword.cpp:
* Source/WebCore/css/color/CSSUnresolvedColorKeyword.h:
* Source/WebCore/css/color/CSSUnresolvedColorLayers.cpp:
* Source/WebCore/css/color/CSSUnresolvedColorLayers.h:
* Source/WebCore/css/color/CSSUnresolvedColorMix.cpp:
* Source/WebCore/css/color/CSSUnresolvedColorMix.h:
* Source/WebCore/css/color/CSSUnresolvedColorResolutionState.cpp: Renamed from Source/WebCore/css/color/CSSUnresolvedColorResolutionContext.cpp.
* Source/WebCore/css/color/CSSUnresolvedColorResolutionState.h: Renamed from Source/WebCore/css/color/CSSUnresolvedColorResolutionContext.h.
* Source/WebCore/css/color/CSSUnresolvedLightDark.cpp:
* Source/WebCore/css/color/CSSUnresolvedLightDark.h:
* Source/WebCore/css/color/CSSUnresolvedRelativeColor.h:
* Source/WebCore/css/color/CSSUnresolvedStyleColorResolutionState.h: Copied from Source/WebCore/style/ColorFromPrimitiveValue.h.
* Source/WebCore/css/color/StyleColor.cpp:
* Source/WebCore/css/color/StyleColor.h:
* Source/WebCore/css/color/StyleColorLayers.cpp:
* Source/WebCore/css/color/StyleColorLayers.h:
* Source/WebCore/css/color/StyleColorMix.cpp:
* Source/WebCore/css/color/StyleColorMix.h:
* Source/WebCore/css/color/StyleRelativeColor.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Color.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Color.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+RawResolver.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+UnevaluatedCalc.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+UnevaluatedCalc.h:
* Source/WebCore/html/canvas/CanvasStyle.cpp:
* Source/WebCore/platform/calc/CalculationTree+Copy.cpp:
* Source/WebCore/style/ColorFromPrimitiveValue.cpp:
* Source/WebCore/style/ColorFromPrimitiveValue.h:
* Source/WebCore/style/FilterOperationsBuilder.cpp:
* Source/WebCore/style/StyleBuilderState.cpp:

Canonical link: https://commits.webkit.org/282670@main
